apps/prairielearn/src/sync/course-db.ts

```diff
@@ -1434,6 +1408,165 @@ function validateCourseInstance({
   return { warnings, errors };
 }
 
+export function validateAccessControlArray({
```

Comment:
You can use `AccessControlJson`. The validation functions run after Zod parsing (which would turn `AccessControlJsonInput` into `AccessControlJson`)